### PR TITLE
embind: Remove str-to-func eval of named function factory

### PR DESCRIPTION
**Pull request by sbc100**
For support for this feature see:
https://caniuse.com/mdn-javascript_operators_object_initializer_computed_property_names

Based on #17296

### CODE REVIEW

File: src/embind/embind.js [hunk 1]
```diff
@@ -198,21 +198,11 @@ var LibraryEmbind = {
...
-    return new Function(
-        "body",
-        "return function " + name + "() {\n" +
-        "    \"use strict\";" +
-        "    return body.apply(this, arguments);\n" +
-        "};\n"
-    )(body);
-#endif
+    return {
+      [name]: function () {
```
Comment (kripken) [inline src/embind/embind.js:202]: Maybe add a comment here? I guess this is using some clever JS VM behavior of guessing the function name based on the property it is written to..?

Comment (kripken) [inline src/embind/embind.js:202]:
```suggestion
      [name]: function() {
```

Comment (sbc100) [reply src/embind/embind.js]: Yes, its a pretty clever solution.  We have tests that verify that it does indeed work (as your can see from the test expectations update).   Added a comment.

**Review by kripken — APPROVED** [review]
Nice!

**Pull request merged**